### PR TITLE
safely parse secretString

### PR DESCRIPTION
**Pull request by simontegg**

### CODE REVIEW

**Comment by simontegg** [comment]
fixes #330

**Comment by lmammino** [comment]
This pull request **introduces 1 alert** when merging d447ac9795f274841803d058b03977dde72a5c54 into 7fdee7876c095f1e1f58fe2966199da2f63fb514 - [view on LGTM.com](https://lgtm.com/projects/g/middyjs/middy/rev/pr-96360f949637e22081e66920b7e198b2d4b096e3)

**new alerts:**

* 1 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/middlewares/secretsManager.js [hunk 1]
```diff
@@ -111,3 +111,13 @@ const getSecretsManagerInstance = awsSdkOptions => {
   const { SecretsManager } = require('aws-sdk')
   return new SecretsManager(awsSdkOptions)
 }
+
+function safeParse (secretString) {
+  try {
+    return JSON.parse(secretString || '{}')
+  } catch (err) {
+    console.info('plain string')
```
Comment (theburningmonk) [inline src/middlewares/secretsManager.js]: how about `return secretString` here?

Comment (lmammino) [reply src/middlewares/secretsManager.js]: Agreed, I would leave the `catch` block empty and return the original unparsed string.

As an alternative, which would add a little bit more complexity, we could expose the functionality to handle errors, by giving the user the possibility to inject a function in the config that will be called here with the given error and the given original string.

**Comment by lmammino** [comment]
This is great stuff, thanks @simontegg!

I added a comment. Also it would be nice to have some test to cover this new use case (simulating a secret in invalid JSON format).

**Comment by lmammino** [comment]
Hello @simontegg, any update on this?

**Comment by simontegg** [comment]
@lmammino sorry for no update. We launched last week! I can get to this over the weekend

**Pull request merged**